src/eq.cpp

```diff
@@ -62,25 +78,60 @@ void  MainFrame::designEQFilters(paCallBackData *cb, int rxSampleRate, int txSam
     // init Spk Out Equaliser Filters
 
     if (m_newSpkOutFilter) {
+        assert(cb->sbqSpkOutBass == nullptr && cb->sbqSpkOutTreble == nullptr && cb->sbqSpkOutMid == nullptr);
         //printf("designing new Spk Out filters\n");
         //printf("designEQFilters: wxGetApp().m_SpkOutBassFreqHz: %f\n",wxGetApp().m_SpkOutBassFreqHz);
         cb->sbqSpkOutBass   = designAnEQFilter("bass", wxGetApp().m_SpkOutBassFreqHz, wxGetApp().m_SpkOutBassGaindB, rxSampleRate);
         cb->sbqSpkOutTreble = designAnEQFilter("treble", wxGetApp().m_SpkOutTrebleFreqHz, wxGetApp().m_SpkOutTrebleGaindB, rxSampleRate);
         cb->sbqSpkOutMid    = designAnEQFilter("equalizer", wxGetApp().m_SpkOutMidFreqHz, wxGetApp().m_SpkOutMidGaindB, wxGetApp().m_SpkOutMidQ, rxSampleRate);
     }
+    
+    m_newMicInFilter = false;
+    m_newSpkOutFilter = false;
 }
 
 void  MainFrame::deleteEQFilters(paCallBackData *cb)
 {
-    if (m_newMicInFilter) {
-        sox_biquad_destroy(cb->sbqMicInBass);
-        sox_biquad_destroy(cb->sbqMicInTreble);
-        sox_biquad_destroy(cb->sbqMicInMid);
+    if (m_newMicInFilter) 
+    {
+        if (cb->sbqMicInBass) 
+        {
```

Comment:
As above - is it possible to have just one of these filters null, and the others non-null?

Reply:
Can all these changes can be reverted too, simplifying the code?

Reply:
Yep, will do.